src/elexmodel/utils/math_utils.py

```diff
@@ -55,11 +65,23 @@ def weighted_median(x, weights):
     return x_sorted[median_index + 1]
 
 
-def boot_sigma(data, conf, num_iterations=10000):
+def robust_sample_std(x, axis):
+    """
+    Compute the robust sample standard deviation along the last axis by calling winsorize_std.
+    """
+    return winsorize_std(x, axis=-1)
+
+
+def boot_sigma(data, conf, num_iterations=10000, winsorize=winsorize):
     """
     Bootstrap standard deviation.
     """
     # we use upper bound of confidence interval for more robustness
+
+    if winsorize:
+        return bootstrap(
+            data.reshape(1, -1), robust_sample_std, confidence_level=conf, method="basic", n_resamples=num_iterations
+        ).confidence_interval.high
```

Comment:
I think this can be simplified a bit by defining a variable called `std_func` which is dependent on whether `winsorize` is true or not, which then gets passed into `bootstrap` either way